pwt watch does not show error when loaders missing

Delete a loader from node modules (or don't install it). The build error messages do not get surfaced?

```
Building webpack.client.dev...
[HPM] Proxy created: /  ->  http://localhost:3101
(node:47172) DeprecationWarning: Tapable.plugin is deprecated. Use new API on `.hooks` instead
Type checking in progress...
module.js:550
    throw err;
    ^

Error: Cannot find module '/Users/jakeg/_code/web/src/the-west/dist/server.js'
    at Function.Module._resolveFilename (module.js:548:15)
    at Function.Module._load (module.js:475:25)
    at Function.Module.runMain (module.js:694:10)
    at startup (bootstrap_node.js:204:16)
    at bootstrap_node.js:625:3
```

--- a/docs/build.md
+++ b/docs/build.md
@@ -96,7 +96,7 @@ export default config
 
 ## Running
 
-    pwt start [watch] [fast] [debug] [prod]
+    pwt start [watch] [debug] [prod]
 
 Starts the server, using the environment variables defined in `.env`
 
@@ -104,7 +104,6 @@ Starts the server, using the environment variables defined in `.env`
 
 -   `NODE_ENV`: set to `"production"` or `"development` depending on the `prod` flag - in the webpack bundles the values are hardcoded depending on which mode they were built in, so this is only relevant for third-party dependencies.
 -   `START_WATCH_MODE`: set to `"true"` by the `watch` flag
--   `START_FAST_MODE`: set to `"true"` by the `fast` flag
 -   `START_DEBUG_MODE`: set to `"true"` by the `debug` flag
 
 If you want to use additional `process.env` variables in the **client** build, make sure you create a `.env` (for local) and `.env.default` file:

--- a/docs/cli.md
+++ b/docs/cli.md
@@ -36,11 +36,10 @@ Opens the `webpack-bundle-analyzer` for the client production bundle.
 ### start
 
 ```
-pwt start [watch] [fast] [prod] [debug] [client] [-p <project dir>]
+pwt start [watch] [prod] [debug] [client] [-p <project dir>]
 ```
 
 -   `watch`: Enable watch mode and rebuild client after changes
--   `fast`: Disable TypeScript type checking for faster incremental builds
 -   `prod`: Set `NODE_ENV` to `"production"`
 -   `debug`: Start node process with `--inspect` for debugging and performance measurement in _chrome://inspect_
 -   `client`: Start internal server that serves only the client
@@ -102,12 +101,11 @@ On build servers with access to Google Chrome through a Docker container, the fo
 ### watch
 
 ```
-pwt watch [server] [client] [fast] [debug] [-p <project dir>]
+pwt watch [server] [client] [debug] [-p <project dir>]
 ```
 
 Builds the server in dev mode, then watches and rebuilds the client
 
 -   `server`: Also watches and rebuilds the server
--   `fast`: Disable TypeScript type checking for faster incremental builds
 -   `client`: Only run client without a server
 -   `debug`: Start node process with `--inspect` for debugging and performance measurement in _chrome://inspect_

--- a/lib/bin/_pwt.ts
+++ b/lib/bin/_pwt.ts
@@ -13,14 +13,6 @@ const env: NodeJS.ProcessEnv = {
     ...process.env,
 }
 
-// already set fast mode env variable here so that the server build also skips type checking
-if (
-    (process.argv.indexOf('watch') !== -1 && process.argv.indexOf('fast') !== -1) ||
-    process.argv.indexOf('build') !== -1
-) {
-    env.START_FAST_MODE = 'true' // fast mode when building
-}
-
 const proc = fork(filePath, process.argv.slice(2), {
     env,
     execArgv: [

--- a/lib/bin/index.ts
+++ b/lib/bin/index.ts
@@ -74,9 +74,9 @@ Scripts:
     build [<target>] [<environment>] [-p <project dir>]
     clean [-p <project dir>] [<glob> ...]
     explore-bundle [disableHoisting] [-p <project dir>]
-    start [watch] [fast] [prod] [debug] [client] [-p <project dir>]
+    start [watch] [prod] [debug] [client] [-p <project dir>]
     stats [verbose] [-p <project dir>]
-    watch [server] [client] [fast] [debug] [-p <project dir>]
+    watch [server] [client] [debug] [-p <project dir>]
 
 Refer to docs/cli.md for the full API documentation
 `)

--- a/lib/bin/start.ts
+++ b/lib/bin/start.ts
@@ -11,7 +11,6 @@ import { forkPromise } from '../util/process'
  * defined in `.env`
  * @param args
  * - watch: Enables watch mode
- * - fast: Disables server-side rendering and type checking
  * - prod: Sets NODE_ENV to "production"
  * - debug: Starts with debugging enabled
  * - debug-brk: Starts with debugging and waits for debugger to attach
@@ -36,10 +35,6 @@ const start = (
 
     if (args.indexOf('watch') !== -1) {
         env.START_WATCH_MODE = 'true'
-    }
-
-    if (args.indexOf('fast') !== -1) {
-        env.START_FAST_MODE = 'true'
     }
 
     const isDebug = args.indexOf('debug') !== -1

--- a/lib/bin/watch.ts
+++ b/lib/bin/watch.ts
@@ -11,7 +11,6 @@ import start from './start'
  * Rebuilds the client on changes
  * @param args
  * - server: Also watches and rebuilds server
- * - fast: disables type checking
  * - client: Only run client without a server
  */
 const watch = async (
@@ -23,11 +22,6 @@ const watch = async (
     const { HAS_SERVER } = buildConfig
     const additionalStartParams: StartParam[] = []
     const env: NodeJS.ProcessEnv = { ...watchProcessEnv }
-
-    if (args.indexOf('fast') !== -1) {
-        additionalStartParams.push('fast')
-        env.START_FAST_MODE = 'true'
-    }
 
     await clean(log, buildConfig)
 

--- a/lib/config/ts-loader-config.ts
+++ b/lib/config/ts-loader-config.ts
@@ -1,23 +1,9 @@
-import os from 'os'
 import webpack from 'webpack'
 import TsconfigPathsPlugin from 'tsconfig-paths-webpack-plugin'
-import ForkTsCheckerWebpackPlugin from 'fork-ts-checker-webpack-plugin'
 import { CreateWebpackConfigOptions } from '../runtime/server'
 import { BuildTarget } from '../types'
 
 const disableCaching = String(process.env.BUILD_CACHE_DISABLED).toLowerCase() === 'true'
-const disableTypeCheck = process.env.START_FAST_MODE === 'true'
-
-const osCpus = os.cpus().length
-const threadLoaderCpus = !disableTypeCheck ? osCpus - 1 : osCpus
-
-const threadLoader = {
-    loader: 'thread-loader',
-    options: {
-        // there should be 1 cpu for the fork-ts-checker-webpack-plugin if its enabled
-        workers: threadLoaderCpus,
-    },
-}
 
 const cacheLoader = (cacheDirectory: string) => {
     return {
@@ -29,25 +15,20 @@ const cacheLoader = (cacheDirectory: string) => {
 }
 
 export function getTypeScriptWebpackRule(
-    plugins: webpack.Plugin[],
+    _plugins: webpack.Plugin[],
     resolvePlugins: webpack.ResolvePlugin[],
     options: CreateWebpackConfigOptions,
     buildTarget: BuildTarget,
 ): webpack.RuleSetRule {
-    if (!disableTypeCheck) {
-        plugins.push(new ForkTsCheckerWebpackPlugin({ checkSyntacticErrors: true })) // fork ts checking
-    }
-
     const configFile =
-        buildTarget === 'server'
+        (buildTarget === 'server'
             ? options.buildConfig.TS_CONFIG_SERVER
-            : options.buildConfig.TS_CONFIG_CLIENT
+            : options.buildConfig.TS_CONFIG_CLIENT) || 'tsconfig.json'
+
+    options.log.info(`Target ${buildTarget} using ts config file: ${configFile}`)
 
     const tsConfigPathsPluginConfig: any = {}
-    const tsLoaderOptions: any = {
-        happyPackMode: true, // IMPORTANT! use happyPackMode mode to speed-up compilation and reduce errors reported to webpack
-        transpileOnly: true,
-    }
+    const tsLoaderOptions: any = {}
 
     if (configFile) {
         tsConfigPathsPluginConfig.configFile = configFile
@@ -62,8 +43,6 @@ export function getTypeScriptWebpackRule(
 
     return {
         test: /\.tsx?$/,
-        use: !disableCaching
-            ? [threadLoader, cacheLoader(options.cacheDirectory), tsLoader]
-            : [threadLoader, tsLoader],
+        use: !disableCaching ? [cacheLoader(options.cacheDirectory), tsLoader] : [tsLoader],
     }
 }

--- a/lib/config/ts-loader-config.ts
+++ b/lib/config/ts-loader-config.ts
@@ -28,7 +28,11 @@ export function getTypeScriptWebpackRule(
     options.log.info(`Target ${buildTarget} using ts config file: ${configFile}`)
 
     const tsConfigPathsPluginConfig: any = {}
-    const tsLoaderOptions: any = {}
+    const tsLoaderOptions: any = {
+        compilerOptions: {
+            noEmit: false,
+        },
+    }
 
     if (configFile) {
         tsConfigPathsPluginConfig.configFile = configFile

--- a/lib/runtime/server/server.ts
+++ b/lib/runtime/server/server.ts
@@ -22,8 +22,6 @@ export const getPort = (fallbackPort?: number) => {
 
 export const isWatchMode = () => process.env.START_WATCH_MODE === 'true'
 
-export const isFastMode = () => process.env.START_FAST_MODE === 'true'
-
 export interface CreateServerOptions {
     log: Logger
 

--- a/lib/types/index.ts
+++ b/lib/types/index.ts
@@ -5,7 +5,6 @@ export type BuildEnvironment = 'dev' | 'prod' | 'debug'
 export type BuildParam = BuildTarget | BuildEnvironment
 
 export type StartParam =
-    | 'fast'
     | 'watch'
     | 'prod'
     | 'debug'
@@ -14,4 +13,4 @@ export type StartParam =
     | '--debug-port'
     | number
 
-export type WatchParam = 'server' | 'fast' | 'client'
+export type WatchParam = 'server' | 'client'

--- a/lib/watch/server.ts
+++ b/lib/watch/server.ts
@@ -11,6 +11,7 @@ import { getWebpackConfig } from '../build/build'
 import { getPort } from '../runtime/server/server'
 import { findFreePort, waitForConnection } from '../runtime/util/network'
 import { getHotReloadMiddleware, openBrowser } from '../server/dev'
+import { webpackStatsConfig } from '../util/webpack'
 
 const restartServer = (
     buildConfig: BuildConfig,
@@ -51,15 +52,30 @@ async function watchServer(log: Logger, buildConfig: BuildConfig) {
     let devServer: ChildProcess
     let devServerAvailable: Promise<any>
 
-    const serverCompiler = webpack(getWebpackConfig(log, buildConfig, 'server', 'dev'))
+    const serverConfig = getWebpackConfig(log, buildConfig, 'server', 'dev')
+    const serverCompiler = webpack(serverConfig)
 
     serverCompiler.hooks.invalid.tap('invalid', () => {
         log.info('⭐  Server changed, rebuilding and restarting server...  ⭐')
     })
 
-    const watching = serverCompiler.watch({}, () => {
+    const watching = serverCompiler.watch({}, (err, stats) => {
         if (!devServer) {
             setTimeout(() => openBrowser(log, hostPort), 2000)
+            }
+
+            if (err) {
+                log.error({ err }, 'Failed to compile')
+                return
+            } else {
+                const statsString = stats.toString(webpackStatsConfig)
+
+                log.info(statsString)
+
+                if (stats.hasErrors()) {
+                    log.error('Stats has errors')
+                    return
+                }
         }
         devServer = restartServer(buildConfig, devServerPort, buildConfig.BASE, devServer)
 

